### PR TITLE
Added percent GQ0 metric (per sample) to CollectVariantCallingMetrics

### PR DESCRIPTION
**Pull request by pjfan**
Addresses: https://github.com/broadinstitute/picard/issues/286

Added a new metric to the detail metrics output (the metrics that are calculated per sample) by CollectVariantCallingMetrics. Added the new metrics (TOTAL_GQ0_VARIANTS and PCT_GQ0_VARIANTS) to the existing CollectVariantCallingMetricsTest code. Changed the GQ score to 0 for sample HG00160 for two variants (1:985443  rs71576592 and 1:985449  rs56255212)  in mini.vcf file that the CollectVariantCallingMetricsTest code is run on so that the tested output would not be 0.

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [x] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [x] All tests passing on Travis

#### Review
- [x] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: testdata/picard/vcf/mini.vcf [hunk 1]
```diff
@@ -325,10 +325,10 @@
...
....739;HaplotypeScore=13.0598;InbreedingCoeff=0.1155;MLEAC=60;MLEAF=0.698;MQ=53.87;MQ0=0;MQRankSum=-0.873;QD=11.34;RPA=15,14;RU=G;ReadPosRankSum=-0.534;SNPEFF_EFFECT=DOWNSTREAM;SNPEFF_FUNCTIONAL_CLASS=NONE;SNPEFF_GENE_BIOTYPE=protein_coding;SNPEFF_GENE_NAME=AGRN;SNPEFF_IMPACT=MODIFIER;SNPEFF_TRANSCRIPT_ID=ENST00000466223;STR	GT:AD:DP:GQ:PL	1/1:0,3:5:9:73,9,0	0/0:0,0:4:0:0,0,1	0/0:0,0:3:1:0,1,1	0/1:1,1:2:0:25,0,9	0/0:1,2:3:1:0,1,5	./.	1/1:0,2:3:6:62,6,0	./.	0/0:3,0:3:9:0,9,75	0/1:3,4:9:28:103,0,28	1/1:0,2:2:6:62,6,0	1/1:0,2:3:6:62,6,0	1/1:0,1:1:3:31,3,0	0/1:2,3:11:29:79,0,29	1/1:0,1:6:3:31,3,0	1/1:0,2:6:6:60,6,0	1/1:0,1:1:3:29,3,0	./.	1/1:0,1:1:3:31,3,0	./.	0/0:2,0:2:6:0,6,26	1/1:0,1:5:3:31,3,0	0/0:2,0:3:6:0,6,32	1/1:0,1:2:3:31,3,0	1/1:0,1:2:3:29,3,0	./.	1/1:0,2:6:6:62,6,0	1/1:0,4:8:12:116,12,0	0/1:1,1:4:9:25,0,9	1/1:0,1:1:3:31,3,0	0/0:1,0:7:3:0,3,19	0/1:4,3:9:54:68,0,54	1/1:0,3:3:9:90,9,0	0/1:1,3:9:5:65,0,5	1/1:0,2:5:6:62,6,0	1/1:0,1:4:3:10,3,0	1/1:0,7:12:21:213,21,0	0/1:1,2:5:5:53,0,5	0/0:1,0:1:3:0,3,11	1/1:0,2:2:6:62,6,0	1/1:0,1:3:3:31,3,0	0/0:1,0:2:3:0,3,12	./.	./.	1/1:0,1:2:3:31,3,0	0/1:1,2:4:3:51,0,3	1/1:0,2:2:6:62,6,0	0/1:2,1:3:18:18,0,35	0/1:3,1:6:15:21,0,15	0/0:1,0:1:3:0,3,16
```
Comment (gbggrant) [inline testdata/picard/vcf/mini.vcf]: Please upload a new index file since you've modified mini.vcf

**Comment by pjfan** [comment]
@gbggrant Updated the mini.vcf.idx 

Also (in a separate commit) updated testMetricsTinyGVCF() to also test for the new metrics and fixed the bug (https://github.com/broadinstitute/picard/issues/800) present in testMetricsTinyGVCF() causing tests to auto pass. (Modified the mini_gvcf.vcf and mini_gvcf.vcf.idx too)

**Comment by gbggrant** [comment]
Thanks @pjfan looks good.  👍

**Comment by fleharty** [comment]
@pjfan @gbggrant 
Before we merge this.

I'm concerned that because this adds a column TOTAL_GQ0_VARIANTS that it could break a downstream database used in production.  @gbggrant do you know if this is possible, or something we should worry about?

**Comment by gbggrant** [comment]
@fleharty I spoke with @ktibbett about this and we have tickets ready for this (that is to do the corresponding database work to handle the additional metrics in picard public).

Thanks for checking on this.

**Comment by fleharty** [comment]
@gbggrant Awesome, ty!

**Pull request merged**